src/main/java/com/minecolonies/coremod/colony/buildings/workerbuildings/BuildingBaker.java

```diff
@@ -276,6 +279,12 @@ public void onColonyTick(@NotNull final IColony colony)
     @Override
     public boolean canRecipeBeAdded(final IToken token)
     {
+
+        ResourceLocation builder_products = new ResourceLocation("minecolonies", this.getJobName().toLowerCase().concat("_product"));
```

Comment:
This logic can be abstracted into a protected method in the abstract building: canRecipeBeAddedBasedOnTags(final IToken token) which can then be reused through out the buildings even when the recipe addition logic is changed.

Please rework this as such, since this is a lot of copy pasting.